solidity/random-beacon/contracts/libraries/DKG.sol

```diff
@@ -410,6 +413,15 @@ library DKG {
             "challenge period has not passed yet"
         );
 
+        require(
+            msg.sender == self.resultSubmitter ||
+                block.number >
+                self.submittedResultBlock +
+                    self.parameters.resultChallengePeriodLength +
+                    self.parameters.resultSubmissionEligibilityDelay,
+            "Only the DKG result submitter can approve the result at this moment"
+        );
```

Comment:
Nitpicking be we could extract `self.submittedResultBlock + self.parameters.resultChallengePeriodLength` into a variable. I think it should be a bit cheaper in case the optimizer does not take care about it:

```
uint256 challengePeriodEnd = self.submittedResultBlock + self.parameters.resultChallengePeriodLength;

require(block.number > challengePeriodEnd, ...);

require(msg.sender == self.resultSubmitter || block.number > challengePeriodEnd + self.parameters.resultSubmissionEligibilityDelay, ...);

Reply:
Done in https://github.com/keep-network/keep-core/pull/2709/commits/c226e3798c730dd504a46897d6aeb74d265a4e54.